Some features from Spring Data JPA have been added.

ShastkivRuslan opened the pull request:
**Changes:**

- Replaced the existing BookRepository with one that uses the JpaRepository interface.
- Added Liquibase support for database schema management.
- Changed the spring.jpa.hibernate.ddl-auto property value to validate.
- Implemented the soft delete concept for the Book entity.
- Added missing endpoints and methods for BookService.

**Endpoints in BookController:**

- GET /api/books (Retrieve book catalog)
- GET /api/books/{id} (Retrieve book details)
- POST /api/books (Create a new book)
- PUT /api/books/{id} (Update a specific book)
- DELETE /api/books/{id} (Delete a specific book)

**Note:
The method for updating the Book entity uses the repository's save() method. If the passed entity has an identifier, it will automatically update the entity with the corresponding identifier.**

ShastkivRuslan committed:
Revert "Create README.md"

This reverts commit 81ca7623999967327b9cc7ecdd42ac145ef291bf.

ShastkivRuslan committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	README.md

ShastkivRuslan committed:
Replaced the existing BookRepository

ShastkivRuslan committed:
added liquibase dependencies

ShastkivRuslan committed:
-added liquibase files
-changed name of book entity table
-changed ddl-auto to validate
-added custom name for column coverImage;

ShastkivRuslan committed:
-implemented soft-delete concept for book entity

ShastkivRuslan committed:
Final commit;
Changes:

Replaced the existing BookRepository with one that uses the JpaRepository interface.
Added Liquibase support for database schema management.
Changed the spring.jpa.hibernate.ddl-auto property value to validate.
Implemented the soft delete concept for the Book entity.
Added missing endpoints and methods for BookService.
Endpoints in BookController:

GET /api/books (Retrieve book catalog)
GET /api/books/{id} (Retrieve book details)
POST /api/books (Create a new book)
PUT /api/books/{id} (Update a specific book)
DELETE /api/books/{id} (Delete a specific book)

ShastkivRuslan committed:
bug-fix in property file;

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/model/Book.java:
```diff
@@ -9,23 +9,37 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
+
+    @Column(nullable = false, name = "is_deleted")
```
```suggestion
    @Column(nullable = false)
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/model/Book.java:
```diff
@@ -9,23 +9,37 @@
+
     private String description;
+
+    @Column(name = "cover_image")
```
```suggestion
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/service/impl/BookServiceImpl.java:
```diff
@@ -32,9 +32,24 @@ public List<BookDto> getAll() {
+
+    @Override
+    public BookDto updateBook(Long id, CreateBookRequestDto requestDto) {
+        if (bookRepository.findById(id).isEmpty()) {
+            throw new EntityNotFoundException("Can t find book by id: " + id);
+        }
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(()....
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

Elena-Bruyako commented on src/main/resources/application.properties:
```diff
@@ -1,8 +1 @@
 spring.application.name=bookstore
-spring.datasource.url=jdbc:mysql://localhost:3306/bookstore
```
Why did you remove all these properties?

Elena-Bruyako commented on src/main/resources/db/changelog/changes/01-create-books-table.yaml:
```diff
@@ -0,0 +1,45 @@
+                    nullable: false
+              - column:
+                  name: price
+                  type: decimal
```
add not null constraint

Elena-Bruyako commented on src/main/resources/db/changelog/changes/01-create-books-table.yaml:
```diff
@@ -0,0 +1,45 @@
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
```
add empty line at the end of the file

Elena-Bruyako reviewed (changes requested):
see comments

ShastkivRuslan committed:
Changes after mentor review:
-Removed the name = "is_deleted" attribute from the @Column annotation in Book entity;
-Removed @Column(name = "cover_image") annotation from the coverImage field in Book entity;
-Refactored the updateBook method to streamline the update process.
 --Replaced manual ID check with findBookById helper method for retrieving and validating the existence of a book.
 --Utilized bookMapper.updateBookFromDto to map updates directly onto the existing Book entity.
 --Simplified error handling by centralizing the entity lookup and exception throwing within findBookById.
 -Added NOT NULL constraint to the price column with DECIMAL type to ensure it cannot be null.
 -Added an empty line at the end of changes/01-create-books-table.yaml to comply with file formatting standards.
 -Restored lines in the application.properties file to previous state.

Elena-Bruyako reviewed (approved):
well done
